### PR TITLE
update lint to 1.59.1

### PR DESCRIPTION
**Pull request by edii**

### CODE REVIEW

File: config/.golangci-no-plugin.yml [hunk 1]
```diff
@@ -187,4 +199,4 @@ issues:
       text: "unnecessaryDefer:"
 run:
   timeout: 10m
-  go: '1.20'
\ No newline at end of file
+  go: '1.22'
```
Comment (trapck) [inline config/.golangci-no-plugin.yml:202]: add an empty line

File: config/.golangci.yml [hunk 2]
```diff
@@ -194,4 +206,4 @@ issues:
 
 run:
   timeout: 10m
-  go: '1.20'
\ No newline at end of file
+  go: '1.22'
```
Comment (trapck) [inline config/.golangci.yml:209]: add an empty line

File: config/.golangci-no-plugin.yml (hunk 1 above)
Comment (edii) [reply config/.golangci-no-plugin.yml]: Done

File: config/.golangci.yml (hunk 2 above)
Comment (edii) [reply config/.golangci.yml]: Done

**Review by trapck — APPROVED** [review]

**Pull request merged**